stagecraft/apps/collectors/views.py

```diff
@@ -0,0 +1,329 @@
+from django.utils.decorators import method_decorator
+from django.views.decorators.cache import never_cache
+from django.views.decorators.vary import vary_on_headers
+from stagecraft.apps.collectors.models import Provider, DataSource, \
+    CollectorType, Collector
+from stagecraft.apps.datasets.models import DataSet
+from stagecraft.libs.authorization.http import _get_resource_role_permissions
+from stagecraft.libs.views.resource import ResourceView, UUID_RE_STRING
+from stagecraft.libs.views.utils import create_http_error
+import logging
+
+logger = logging.getLogger(__name__)
+logger.setLevel('WARNING')
+
+
+class ProviderView(ResourceView):
+    model = Provider
+
+    schema = {
+        "$schema": "http://json-schema.org/schema#",
+        "type": "object",
+        "properties": {
+            "name": {
+                "type": "string"
+            },
+            "credentials_schema": {
+                "type": "object"
+            }
+        },
+        "required": ["name"],
+        "additionalProperties": False,
+    }
+
+    id_fields = {
+        'id': UUID_RE_STRING,
+        'name': '[\w-]+',
+    }
+    list_filters = {
+        "name": "name__iexact"
+    }
+
+    permissions = _get_resource_role_permissions('Provider')
```

Comment:
Similarly to the decorators, this could be pushed down into the init.  As it appears to be the same on all views.